src/commands/generate/__tests__/tsReturnTypeVisitorTest.ts

```diff
@@ -17,169 +17,269 @@
 
 import { IType, ITypeDefinition, PrimitiveType } from "conjure-api";
 import { TsReturnTypeVisitor } from "../tsReturnTypeVisitor";
-import { ITypeGenerationFlags } from "../typeGenerationFlags";
 import { createHashableTypeName } from "../utils";
+import { DEFAULT_TYPE_GENERATION_FLAGS, FLAVORED_TYPE_GENERATION_FLAGS, READONLY_COLLECTION_TYPE_GENERATION_FLAGS } from "./resources/constants";
 
-const GENERATION_FLAGS_TO_USE: ITypeGenerationFlags = {
-    flavorizedAliases: true,
-};
+const objectName = { name: "Object", package: "" };
+const objectReference = IType.reference(objectName);
+const object = ITypeDefinition.object({
+    fields: [],
+    typeName: objectName,
+});
+
+const aliasName = { name: "Alias", package: "" };
+const aliasReference = IType.reference(aliasName);
+const alias = ITypeDefinition.alias({
+    alias: { primitive: PrimitiveType.STRING, type: "primitive" },
+    typeName: aliasName,
+});
+
+const binaryAliasName = { name: "BinaryAlias", package: "" };
+const binaryAliasReference = IType.reference(binaryAliasName);
+const binaryAlias = ITypeDefinition.alias({
+    alias: { primitive: PrimitiveType.BINARY, type: "primitive" },
+    typeName: binaryAliasName,
+});
+
+const enumName = { name: "Enum", package: "" };
+const enumReference = IType.reference(enumName);
+const enumType = ITypeDefinition.enum_({
+    typeName: enumName,
+    values: [{ value: "FOO" }],
+});
+
+const fakeTypeName = { name: "someObject", package: "com.palantir.example" };
 
 describe("TsTypeVisitor", () => {
-    const objectName = { name: "Object", package: "" };
-    const objectReference = IType.reference(objectName);
-    const object = ITypeDefinition.object({
-        fields: [],
-        typeName: objectName,
-    });
 
-    const aliasName = { name: "Alias", package: "" };
-    const aliasReference = IType.reference(aliasName);
-    const alias = ITypeDefinition.alias({
-        alias: { primitive: PrimitiveType.STRING, type: "primitive" },
-        typeName: aliasName,
-    });
+    describe("with default generation flags", () => {
```

Comment:
I added an extra level here to group tests with different generation flags, restoring tests for default flags (ie `flavoredAliases: false`) and adding the explicit flavoredAliases & readonly collections cases.
Makes for a terrible diff though without whitespace ignored.